web_editor/js/main.js

```diff
@@ -6,17 +6,32 @@ import init, {
 } from "./osm2lanes-npm/osm2lanes_npm.js";
 
 await init();
-setupOnce();
+document.getElementById("start-editing").onclick = async function () {
+  try {
+    window.app = await LaneEditor.create();
+    window.app.render();
+  } catch (err) {
+    window.alert(`Error: ${err}`);
+  }
+};
 
 export class LaneEditor {
-  constructor(way, road_wrapper, locale, tags) {
-    const road = road_wrapper.Ok.road;
-
+  constructor(way, road, locale, tags) {
     this.way = way;
-    // Clone these
-    this.currentRoad = JSON.parse(JSON.stringify(road));
-    this.currentLocale = JSON.parse(JSON.stringify(locale));
+    this.road = road;
+    this.locale = locale;
     this.originalTags = tags;
+
+    // Remove these immediately. If they're rendered invisibly, they mess up indices.
+    this.road.lanes = this.road.lanes.filter(
+      (lane) => lane.type != "separator"
+    );
+
+    // All of the state is immutable, except for road. The source of truth for
```

Comment:
This is different -- and simpler -- than before, when the source of truth was the sortable cards in the DOM. Anytime we make a mutation, we just re-render all of the cards from scratch. This is cheap enough.

Reply:
The simplicity gained by this approach is a huge win in my experience.